halyard-config/src/main/java/com/netflix/spinnaker/halyard/config/services/v1/GenerateService.java

```diff
@@ -1,4 +1,103 @@
+import com.netflix.spinnaker.halyard.config.config.v1.FileWriter;
+import com.netflix.spinnaker.halyard.config.errors.v1.HalconfigException;
+import com.netflix.spinnaker.halyard.config.model.v1.node.Node;
+import com.netflix.spinnaker.halyard.config.model.v1.node.NodeFilter;
+import com.netflix.spinnaker.halyard.config.model.v1.node.NodeReference;
+import com.netflix.spinnaker.halyard.config.model.v1.problem.Problem;
+import com.netflix.spinnaker.halyard.config.model.v1.problem.ProblemBuilder;
+import com.netflix.spinnaker.halyard.config.spinnaker.v1.ComponentName;
+import com.netflix.spinnaker.halyard.config.spinnaker.v1.ComponentProfile;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import org.yaml.snakeyaml.Yaml;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@Component
+@Slf4j
 public class GenerateService {
+  @Autowired
+  ComponentProfile componentProfile;
+
+  @Autowired
+  LookupService lookupService;
+
+  @Autowired
+  String spinnakerOutputPath;
+
+  @Autowired
+  FileWriter yamlWriter;
+
+  @Autowired
+  Yaml yamlParser;
+
+  @Autowired
+  ObjectMapper objectMapper;
+
+  public void generateConfig(NodeReference nodeReference) {
+    NodeFilter filter = NodeFilter.makeAcceptAllFilter()
+        .extendReference(nodeReference);
+
+    List<Node> matching = lookupService.getMatchingNodesOfType(filter, Node.class);
+
+    Map<ComponentName, List<Node>> configInjectMap = new HashMap<>();
+    for (Node node : matching) {
+      for (ComponentName componentName : node.registeredSpinnakerComponents()) {
+        log.trace("Registering " + node.getNodeName() + " with component " + componentName);
+
+        List<Node> injectable = configInjectMap.getOrDefault(componentName, new ArrayList<>());
+        injectable.add(node);
+        configInjectMap.put(componentName, injectable);
+      }
+    }
+
+    for (ComponentName componentName : ComponentName.values()) {
+      String path = spinnakerOutputPath + componentName.getProfile();
+      log.info("Writing profile to  " + path);
+      try {
+        boolean append = false;
+
+        for (Node config : configInjectMap.getOrDefault(componentName, new ArrayList<>())) {
+          yamlWriter.write(yamlParser.dump(objectMapper.convertValue(config, Map.class)), path, append);
+          append = true;
+        }
+
+        yamlWriter.write("\n### Fetched config begins here\n", path, append);
+
+        String componentYaml = componentProfile.getProfile(componentName);
+
+        yamlWriter.write(componentYaml, path, append);
+      }
+      catch (IOException ioe) {
+        throw new HalconfigException(
+            new ProblemBuilder(Problem.Severity.FATAL,
+                "Failed to write config for component " + componentName + " to path \"" + path + "\"").build()
```

Comment:
what happens to the file backing yamlWriter? It is corrupted right? Does that matter?